erts/emulator/test/hello_SUITE_data/hello.erl

```diff
@@ -0,0 +1,2762 @@
+    end.
+
+swapping() ->
+    Args = get_args8(?FUNCTION_NAME),
+    {A,B,C,D,E,F,G,H} = Args,
+    {B,C,D,E,F,G,H,A} = swapping_1(1, 2, 3, A, B, C, D, E, F, G, H),
+    {B,A,F,D,E,C,H,G} = swapping_2(1, 2, 3, A, B, C, D, E, F, G, H),
+    {A,B,C,D,E,H,G,F} = swapping_3(1, 2, 3, A, B, C, D, E, F, G, H).
+
+swapping_1(X, Y, Z, A, B, C, D, E, F, G, H) ->
+    swapping_build(X, Y, Z, B, C, D, E, F, G, H, A).
+
+swapping_2(X, Y, Z, A, B, C, D, E, F, G, H) ->
+    swapping_build(X, Y, Z, B, A, F, D, E, C, H, G).
+
+swapping_3(X, Y, Z, A, B, C, D, E, F, G, H) ->
+    swapping_build(X, Y, Z, A, B, C, D, E, H, G, F).
+
+swapping_build(_, _, _, A, B, C, D, E, F, G, H) ->
+    {A,B,C,D,E,F,G,H}.
+
+get_args(Tag) ->
+    get_args6(Tag).
+
+get_args6(Tag) ->
+    {{Tag,a},{Tag,b},{Tag,make_ref()},[Tag,d],{Tag,e},[Tag,f]}.
+
+get_args8(Tag) ->
+    list_to_tuple([{Tag,1},{Tag,2}|tuple_to_list(get_args6(Tag))]).
+
+reductions(L0) ->
+    L1 = red(L0, L0, []),
+    L2 = red(L0, L1, []),
+    L2 = rev(rev(L2)),
+    ok.
+
+red([_|T], L, Acc) ->
+    red(T, L, L++Acc);
+red([], _, Acc) ->
+    Acc.
+
+apply_test() ->
+    Mod = id(?MODULE),
+    Id = id(id),
+    {a,b} = Mod:Id({a,b}),
+    {x,y} = apply_fixed_last(Mod, Id, {x,y}),
+
+    Args = id([{a,b,c}]),
+    {a,b,c} = apply(Mod, Id, Args),
+    {a,b,c} = apply_var_last(Mod, Id, Args),
+    {a,b,c} = apply_var_only(Mod, Id, Args),
+
+    ok.
+
+apply_fixed_last(M, F, Arg) ->
+    M:F(Arg).
+
+apply_var_last(M, F, Args) ->
+    _ = id(42),
+    apply(M, F, Args).
+
+apply_var_only(M, F, Args) ->
+    apply(M, F, Args).
+
+guard_bif_test() ->
+    P = id(3.14),
+    Bin = id(<<1,2,3,4,5,6>>),
+    3 = floor(P),
+
+    T = id({a,b,c}),
+    b = element(2, T),
+
+    <<3,4,5>> = binary_part(Bin, 2, 3),
+
+    Tuple = id(get_args8(?FUNCTION_NAME)),
+    if element(1, Tuple) =:= {?FUNCTION_NAME,1} -> ok end,
+
+    gbif_preserving_xregs(Tuple),
+    gbif_nofail(Tuple).
+
+%% Test that all X registers are preserved when calling a guard BIF.
+gbif_preserving_xregs(Tuple) ->
+    X = id(-42),
+    {A,B,C,D,E,F,G,H} = id(Tuple),
+
+    %% We must prevent the compiler from delaying the extraction
+    %% of the tuple elements in the tuple matching above by using
+    %% all extracted elements.
+    1 = element(2, A),
+    2 = element(2, B),
+    a = element(2, C),
+    b = element(2, D),
+    2 = tuple_size(E),
+    d = hd(tl(F)),
+    2 = tuple_size(G),
+    f = hd(tl(H)),
+
+    if
+        %% Test that guard BIFs called in a guard preserve all X registers.
+        abs(X) =:= 42 ->
+            %% Rebuild the tuple from the extracted X registers and match.
+            Tuple = {A,B,C,D,E,F,G,H}
+    end,
+
+    ok.
+
+%% Test guard BIFs that can't fail.
+gbif_nofail(Tuple0) ->
+    Tuple = id(Tuple0),
+
+    true = id(is_tuple(Tuple)),
+    false = id(is_list(Tuple)),
+
+    true = id(Tuple0 =:= Tuple),
+    false = id(Tuple =:= 42),
+    ok.
+
+test_catches() ->
+    thrown = (catch do_throw(id(thrown))),
```

Comment:
I'm sure you know this so feel free to ignore me, but after https://github.com/erlang/otp/pull/4736 gets merged you can drop the parentheses (4 instances of this in the file).

Reply:
Removed the parentheses. (#4736 is not needed because it fixes pretty-printing and documentation, while the actual change to `erl_parse` was done in a previous commit.)